Dockerfile

```diff
@@ -1,66 +1,134 @@
+#RUN dpkg --add-architecture armhf && \
+#    apt update && \
+#    apt install -y gcc-arm-linux-gnueabihf g++-arm-linux-gnueabihf libc6-armhf-cross libc6-dev-armhf-cross libssl-dev:armhf && \
+#    apt clean && \
+#    rm -rf /var/lib/apt/lists/*
+### add armhf target
+#RUN rustup target add rustup target add armv7-unknown-linux-gnueabihf
 
-# Build dependencies.
-FROM chef AS builder
-COPY --from=planner /tmp/recipe.json recipe.json
-RUN cargo chef cook --release -p lldap_app --target wasm32-unknown-unknown \
-    && cargo chef cook --release -p lldap \
-    && cargo chef cook --release -p migration-tool
 
-# Copy the source and build the app and server.
-COPY --chown=app:app . .
-RUN cargo build --release -p lldap -p migration-tool \
-    # Build the frontend.
-    && ./app/build.sh
+# Install cargo-chef and wasm-pack via cargo [npm dont have arm64 bin]    
+RUN RUSTFLAGS="-Ctarget-feature=-crt-static" cargo install cargo-chef wasm-pack
+RUN rustup target add wasm32-unknown-unknown
+# Install rollup
+RUN npm install -g rollup
 
-# Final image
-FROM alpine:3.14
 
+# Prepare dependencies
+FROM builder-base AS planner
 WORKDIR /app
+COPY . .
+RUN cargo chef prepare --recipe-path /tmp/recipe.json
+
+# Build depedencies
+FROM builder-base AS builder
+COPY --from=planner /tmp/recipe.json recipe.json
+RUN RUSTFLAGS="-Ctarget-feature=-crt-static" cargo chef cook --release -p lldap_app --target wasm32-unknown-unknown
+RUN RUSTFLAGS="-Ctarget-feature=-crt-static" cargo chef cook --release -p lldap
+RUN RUSTFLAGS="-Ctarget-feature=-crt-static" cargo chef cook --release -p migration-tool
+COPY . /lldap-src
+WORKDIR /lldap-src
 
-COPY --from=builder /app/app/index_local.html app/index.html
-COPY --from=builder /app/app/static app/static
-COPY --from=builder /app/app/pkg app/pkg
-COPY --from=builder /app/target/release/lldap /app/target/release/migration-tool ./
-COPY docker-entrypoint.sh lldap_config.docker_template.toml ./
+# Compiling application, take your time
+### amd64
+RUN cargo build --target=x86_64-unknown-linux-gnu --release -p lldap -p migration-tool
+#######################################################
+### Only enable if building non-native architecture ###
+#######################################################
+### arm64
+#RUN cargo build --target=aarch64-unknown-linux-gnu --release -p lldap -p migration-tool
+### armhf
+#RUN cargo build --target=armv7-unknown-linux-gnueabihf --release -p lldap -p migration-tool
 
+
+### Build frontend
+RUN app/build.sh
+# Prepare our application path
+RUN mkdir -p /lldap/app
+# Web and App dir
+COPY docker-entrypoint.sh /docker-entrypoint.sh
+COPY lldap_config.docker_template.toml /lldap/
+# The applications
+RUN cp target/*/release/lldap /lldap/lldap && \
+    cp target/*/release/migration-tool /lldap/migration-tool && \
+    cp -R app/index.html \
+          app/pkg \
+          app/static \
+          /lldap/app/
+# Just checking
+RUN ls -al /lldap && \
```

Comment:
Umm, why? We have a cp just above that copies to /lldap/app, so it'll always exist.